CMakeLists.txt

```diff
@@ -272,3 +272,12 @@ if (BENCHMARK_ENABLE_TESTING)
   endif()
   add_subdirectory(test)
 endif()
+
+set(CPACK_GENERATOR "DEB")
+set(CPACK_DEBIAN_PACKAGE_MAINTAINER "Google Inc.")
+set(CPACK_PACKAGE_VERSION ${VERSION})
+set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "Google Benchmark library.")
+set(CPACK_PACKAGE_DESCRIPTION "A library to support the benchmarking of functions, similar to unit-tests.")
+set(CPACK_DEBIAN_PACKAGE_SECTION "Development")
```

Comment:
Should this have any of other DEB magic from https://salsa.debian.org/science-team/benchmark/blob/master/debian/control ?

Reply:
With the CPack approach we're missing that part.

https://salsa.debian.org/science-team/benchmark/blob/master/debian/control is part of the debianize process, which generates a deb file and some additional stuff so the result is compliant with the official Debian rules for packaging. Meaning that things can go into the official repos.

Reply:
I don't know enough about debian and packaging rules. Does this PR as it is allow us to trivially create deb packages and have them in the repo?

Reply:
(i personally hope not, that should be done by actual maintainers, not projects themselves)

Reply:
This allows to trivially create a deb.

It doesn't create a deb that can be used in Debian/Ubuntu apt repos directly though. Neither uploaded to a Launchpad PPA. The reason being that the deb is not enough for those. They need some additional files, e.g. a `.changes` file with Launchpad.

I've tried to run `debuild -us -uc` on the debianized `benchmark` repo from https://salsa.debian.org/science-team/benchmark in Ubuntu 16.04. I had to relax the `debhelper` dependency from `>= 10` to `>= 9` and remove (hacky) the dependency on `libgmock-dev`, which isn't available in Ubuntu 16.04. I still have some errors:
``` bash
$ debuild -us -uc
...
CMake Error at src/CMakeLists.txt:14 (list):
  list does not recognize sub-command FILTER


CMake Error at /usr/share/cmake-3.5/Modules/FindPackageHandleStandardArgs.cmake:148 (message):
  Could NOT find GTest (missing: GTEST_LIBRARY GTEST_MAIN_LIBRARY)
Call Stack (most recent call first):
  /usr/share/cmake-3.5/Modules/FindPackageHandleStandardArgs.cmake:388 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.5/Modules/FindGTest.cmake:205 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  cmake/HandleGTest.cmake:102 (find_package)
  CMakeLists.txt:248 (include)
...
```

And I have `libgtest-dev` version `1.7.0-4ubuntu1` installed in my system.

I guess I could find out how to solve these issues with more time. It's just not worth it for me anymore since with CPack I got a deb we can put in our apt repos and it just works. And once we move to Ubuntu >= 18.04, we can just used the deb officially shipped with it.

So probably, you could just close this PR. :thinking: